phenotrex/ml/prediction.py

```diff
@@ -1,26 +1,44 @@
+import pandas as pd
+
 from phenotrex.io.flat import load_genotype_file, DEFAULT_TRAIT_SIGN_MAPPING
 from phenotrex.io.serialization import load_classifier
+from phenotrex.ml.shap_handler import ShapHandler
+
 try:
     from phenotrex.transforms import fastas_to_grs
 except ModuleNotFoundError:
     from phenotrex.util.helpers import fail_missing_dependency as fastas_to_grs
 
 
-def predict(fasta_files=tuple(), genotype=None, classifier=None, verb=False):
+def predict(fasta_files=tuple(), genotype=None, classifier=None,
+            out_explain_per_sample=None, out_explain_summary=None,
+            shap_n_samples=None, n_max_explained_features=None, verb=False):
     """
     Predict phenotype from a set of (possibly gzipped) DNA or protein FASTA files
-    or a single genotype file.
+    or a single genotype file. Optionally, compute SHAP explanations individually and/or summarily
+    for the predicted samples.
+
     NB: Genotype computation is highly expensive and performed on the fly on FASTA files.
     For increased speed when predicting multiple phenotypes, create a .genotype file to reuse
     with the command `compute-genotype`.
 
+    NB: As opposed to XGB models where they are trivially available, computing SHAP explanations
+    on SVM models entails training a model-agnostic KernelExplainer which is highly costly (dozens
+    to hundreds of seconds per sample if using a somewhat reasonable value for `shap_n_samples`).
+
     :param fasta_files: An iterable of fasta file paths
     :param genotype: A genotype file path
     :param classifier: A pickled classifier file path
+    :param out_explain_per_sample: Where o save the most influential features by SHAP for each
+                                   predicted sample.
+    :param out_explain_summary: Where to save the SHAP summary of the predictions.
+    :param shap_n_samples: The nsamples parameter -
+                               only used by models which utilize a `shap.KernelExplainer`.
```

Comment:
```suggestion
                               only used by models which use a `shap.KernelExplainer`.
```
As I learned from the JMLR authors guide, one should never use 'utilize'. If you want to avoid repetition, perhaps 'incorporate' or similar?

Reply:
interesting, do they give any reason? Sure, I can change it

Reply:
No, full stop :) But the idea is, that 'utilize' implies using in a way that was not intended, or even in direction of 'abuse'.

Reply:
Wow, I never read it like that. Good to know.